replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/SyncHelper.java

```diff
@@ -139,7 +139,24 @@ public SyncHelper(
   }
 
   public SyncResponse sync() {
-    for (Result metacardResult : getMetacardChangeSet()) {
+    // Failed items and new items are replicated separately due to a bug in DDF that was only fixed
+    // recently
+    List<Filter> failedItemFilters = createFailedMetacardFilters();
+    if (!failedItemFilters.isEmpty()) {
+      replicateSubset(builder.anyOf(failedItemFilters));
+    }
+
+    // if the connection was lost while replicating failed items don't continue replicating
+    if (!status.getStatus().equals(Status.CONNECTION_LOST)) {
+      replicateSubset(buildChangeSetFilter());
+    }
+
+    status.setStatus(canceled ? Status.CANCELED : Status.SUCCESS);
+    return new SyncResponse(syncCount, failCount, bytesTransferred, status.getStatus());
+  }
+
+  private void replicateSubset(Filter filter) {
+    for (Result metacardResult : getMetacards(filter)) {
       if (canceled) {
```

Comment:
❓ How about setting the status to `CANCELED`?

Reply:
We set the status to canceled almost immediately after this break. I could set it here if you find that more readable.

Reply:
By doing it here, it is closer to where the decision is made as opposed to be somewhere else along the flow.